Content.Client/Chemistry/UI/HyposprayStatusControl.cs

```diff
@@ -9,34 +11,48 @@ namespace Content.Client.Chemistry.UI;
 
 public sealed class HyposprayStatusControl : Control
 {
-    private readonly HyposprayComponent _parent;
+    private readonly Entity<HyposprayComponent> _parent;
     private readonly RichTextLabel _label;
+    private readonly SharedSolutionContainerSystem _solutionContainers;
 
-    public HyposprayStatusControl(HyposprayComponent parent)
+    private FixedPoint2 PrevVolume;
+    private FixedPoint2 PrevMaxVolume;
+    private bool PrevOnlyAffectsMobs;
+
+    public HyposprayStatusControl(Entity<HyposprayComponent> parent, SharedSolutionContainerSystem solutionContainers)
     {
         _parent = parent;
-        _label = new RichTextLabel {StyleClasses = {StyleNano.StyleClassItemStatus}};
+        _solutionContainers = solutionContainers;
+        _label = new RichTextLabel { StyleClasses = { StyleNano.StyleClassItemStatus } };
         AddChild(_label);
-
-        Update();
     }
 
     protected override void FrameUpdate(FrameEventArgs args)
     {
         base.FrameUpdate(args);
-        if (!_parent.UiUpdateNeeded)
+
+        if (!_solutionContainers.TryGetSolution(_parent.Owner, _parent.Comp.SolutionName, out _, out var solution))
             return;
-        Update();
-    }
 
-    public void Update()
-    {
+        // only updates the UI if any of the details are different than they previously were
+        if (PrevVolume == solution.Volume
+            && PrevMaxVolume == solution.MaxVolume
+            && PrevOnlyAffectsMobs == _parent.Comp.OnlyAffectsMobs)
+            return;
+
```

Comment:
Can't you just dirty it like the old method did and update it once.

Reply:
```protected override void FrameUpdate(FrameEventArgs args)
    {
        base.FrameUpdate(args);
        if (!_parent.UiUpdateNeeded)
           return;
        Update();
    }
```
even though the old method makes use of the component state, isn't it not any different since it still checks the component's status every frame since it makes use of FrameUpdate?

a solution that doesn't seem to rely on checking status every frame is found in Content.Client\Weapons\Ranged\Systems\GunSystem.AmmoCounter.cs, where it relies on the ItemStatusCollectMessage, creates its own control, throws its own event but I have no clue how they determine the contents or style the control at all...

Reply:
#26162 well... I... I'm just going to hope the continued trend of FrameUpdate being used is good enough